src/main/java/jenkins/plugins/logstash/LogstashConfiguration.java

```diff
@@ -171,6 +172,10 @@ public void migrateData()
   @Override
   public boolean configure(StaplerRequest staplerRequest, JSONObject json) throws FormException
   {
+    
+    // Keep a reference to the current configuration to revert back in case of an exception.
+    LogstashIndexer<?> currentIndexer = logstashIndexer;
```

Comment:
maybe `previousIndexer` would be a better name?  "Current" might be understood as referring to the one being processed now